src/docs/development/tools/devtools/code-size.md

```diff
@@ -0,0 +1,231 @@
+relationships that existing between nodes of code size data.
+
+We'll again use this example:
+
+```
+package:a
+|__ package:b
+|   |__ package:d
+|__ package:c
+    |__ package:d
+```
+
+The call graph for this data would link `package:d` to its direct callers,
+`package:b` and `package:c`, instead of its "dominator", `package:a`.
+
+```
+package:a --> package:b -->
+                              package:d
+package:a --> package:c -->
+```
+
+This information is useful for understanding the fine-grained dependencies of
+between pieces of your code (packages, libraries, classes, functions).
+
+![Screenshot of code size call graph]({% asset tools/devtools/code_size_call_graph.png @path %}){:width="100%"}
+
+## Diff tab
+
+The diff tab allows you to compare two snapshots of size information. The two
+size information files you are comparing should be generated from two different
+versions of the same app, for example, the size file generated before and after
+changes to your code. You can visualize the difference between the two data sets
+using the treemap and table.
+
+![Screenshot of code size diff]({% asset tools/devtools/code_size_diff.png @path %}){:width="100%"}
+
+### Loading size files
+
+When you open the Diff tab, you'll see instructions to load "old" and "new" size
+files. Again, these files need to be generated from the same application. Drag
+and drop these files into their respective dialogs, and click "Analyze Diff".
+
+![Screenshot of code size diff loading screen]({% asset tools/devtools/code_size_load_diff.png @path %}){:width="100%"}
+
+See [Generating size files][] below for information on generating these files.
+
+### Treemap and table
+
+In the diff view, the treemap and tree table show only data that differs between
+the two imported size files.
+
+For questions about using the treemap, see [Using the treemap][] above.
+
+## Generating size files
+
+A prerequisite for using the code size tool is generating the proper data to
+analyze. The tool accepts input in a couple different formats.
+
+### Flutter code size analysis files
+
+To analyze the size of your entire app, including analysis for native
+and Dart code, generate a code size analysis file using the `--analyze-size`
+flag:
+
+```
+flutter build <your target platform> --analyze-size
+```
+
+This builds your application, prints a size summary to the command line, and
+prints a line telling you where to find the size analysis file.
+
+```
+flutter build apk --analyze-size --target-platform=android-arm64
+...
+▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒
+app-release.apk (total compressed)                               6 MB
+...
+▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒
+A summary of your APK analysis can be found at: build/apk-code-size-analysis_01.json
+```
+
+In this example, import the `build/apk-code-size-analysis_01.json` file into the
+code size tool to analyze further.
+
+### Dart AOT snapshot files
```

Comment:
Perhaps we can skip this section. I'm not sure if there scenarios where users are better served using this approach since it's a strict subset of everything described above.

Reply:
yeah maybe we won't explicitly say that we support this if we want users to look at the total size anyway